docs/shared/tools-workspace-builders.md

```diff
@@ -0,0 +1,182 @@
+  "type": "object",
+  "properties": {
+    "textToEcho": {
+      "type": "string",
+      "description": "Text To Echo"
+    }
+  }
+}
+```
+
+This example describes a single option for the builder that is a `string` called 'textToEcho’. When using this builder, we'll specify a 'textToEcho' property inside the options.
+
+In our `impl.ts` file, we'll see we're creating an `Options` interface that matches the json object being described here.
+
+### impl.ts
+
+The `impl.ts` contains the actual code for your builder. Your builder should use the `createBuilder` function of the `@angular-devkit/architect` package to create a builder that can be run via the Nx CLI tools.
+
+```ts
+import { BuilderOutput, createBuilder } from '@angular-devkit/architect';
+import * as childProcess from 'child_process';
+import { Observable } from 'rxjs';
+import { json } from '@angular-devkit/core';
+
+interface Options extends json.JsonObject {
+  textToEcho: string;
+}
+
+export default createBuilder((_options: Options, context) => {
+  context.logger.info(`Executing "echo"...`);
+  context.logger.info(`Options: ${JSON.stringify(_options, null, 2)}`);
+  const child = childProcess.spawn('echo', [_options.textToEcho]);
+  return new Observable<BuilderOutput>((observer) => {
+    child.stdout.on('data', (data) => {
+      context.logger.info(data.toString());
+    });
+    child.stderr.on('data', (data) => {
+      context.logger.error(data.toString());
+    });
+    child.on('close', (code) => {
+      context.logger.info(`Done.`);
+      observer.next({ success: code === 0 });
+      observer.complete();
+    });
+  });
+});
+```
+
+See the [official Angular documentation on builders](https://angular.io/guide/cli-builder) for more clarification on creating builders.
+
+Also note that [Node’s `childProcess`](https://nodejs.org/api/child_process.html) is used here which will often be used in most use cases.
+
+Part of the power of the architect API is the ability to compose builders via JS functional composition, where you can combine other builders from your workspace into one which could be helpful when the process you’re scripting is a combination of other existing builders provided by the CLI or other custom-builders in your workspace.
+
+### builder.json
+
+The `builder.json` file provides the description of your builder to the CLI.
+
+```json
+{
+  "builders": {
+    "echo": {
+      "implementation": "./impl",
+      "schema": "./schema.json",
+      "description": "Runs `echo` (to test builders out)."
+    }
+  }
+}
+```
+
+Note that this `builder.json` file is naming our builder 'echo' for the CLI's purposes, and maping that name to the given implemetation file and schema.
+
+### package.json
+
+This is all that’s required from the `package.json` file:
+
+```json
+{
+  "builders": "./builder.json"
+}
+```
+
+## Compiling and Running your Builder
+
+After your files are created, you can compile your builder with `tsc`:
```

Comment:
Maybe add a note here that `tsc` is available if you've installed typescript globally